Dockerfile

```diff
@@ -3,7 +3,7 @@
 
 #################### BASE BUILD IMAGE ####################
 # prepare basic build environment
-FROM nvidia/cuda:12.1.0-devel-ubuntu22.04 AS dev
+FROM nvidia/cuda:12.4.1-devel-ubuntu22.04 AS dev
```

Comment:
12.1.0 is EOL

Reply:
I think cuda has backward compatibility, compiling for 12.1 should work for 12.4 during runtime. But the reverse might not hold. If we compile the wheel with cuda 12.4, not sure if it works for 12.1 during runtime.